### PR TITLE
Add auto-scaling e2e test

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Add e2e test for Auto-scaling. In this e2e test, we test the following cases:
1. Auto scale-out for tikv and check the special hot region label
2. Auto scale-in for tikv and check whether the auto-scaling interval worked.
3. Auto scale-out for tidb
4. Auto scale-in for tidb and check whether the auto-scaling interval worked.

I also add a TODO for mocking the tidbmonitor metrics data in auto-scaling.

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e/tidbcluster/serial.go [hunk 1]
```diff
@@ -713,4 +715,292 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+			// Scale Tikv To 4 replicas and Check
+			tac.Spec.TiKV = &v1alpha1.TikvAutoScalerSpec{
+				BasicAutoScalerSpec: v1alpha1.BasicAutoScalerSpec{
+					MaxReplicas: 4,
+					MinReplicas: pointer.Int32Ptr(4),
+				},
+			}
+			_, err = cli.PingcapV1alpha1().TidbClusterAutoScalers(ns).Create(tac)
+			framework.ExpectNoError(err, "Create TidbMonitorClusterAutoScaler error")
+			pdapi, cancel, err := oa.GetPDClient(tc)
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go:784]: call directly, e.g.  `pdClient, cancel, err := proxiedpdclient.NewProxiedPDClient(c, fw, ns, clusterName, false, nil)`

File: tests/e2e/tidbcluster/serial.go [hunk 2]
```diff
@@ -713,4 +715,292 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+		var ocfg *tests.OperatorConfig
+		var oa tests.OperatorActions
+
+		ginkgo.BeforeEach(func() {
+			ocfg = &tests.OperatorConfig{
+				Namespace:         ns,
+				ReleaseName:       "operator",
+				Image:             cfg.OperatorImage,
+				Tag:               cfg.OperatorTag,
+				SchedulerImage:    "k8s.gcr.io/kube-scheduler",
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go:728]:
```suggestion
```

File: tests/e2e/tidbcluster/serial.go [hunk 3]
```diff
@@ -713,4 +715,292 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+				LogLevel:          "4",
+				ImagePullPolicy:   v1.PullIfNotPresent,
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go:730]:
```suggestion
				LogLevel:          "4",
```

File: tests/e2e/tidbcluster/serial.go (hunk 1 above)
Comment (Yisaer) [reply tests/e2e/tidbcluster/serial.go]: updated.

File: tests/e2e/tidbcluster/serial.go [hunk 4]
```diff
@@ -713,4 +716,290 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+				if !ok {
+					klog.Infof("tac haven't marked any annotation")
+					return false, nil
+				}
+				secondTs, err := strconv.ParseInt(v, 10, 64)
+				if err != nil {
+					return false, err
+				}
+				if secondTs-firstScaleTimestamp < 100 {
+					klog.Infof("tikv second scale's interval isn't meeting the interval requirement")
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/serial.go:903]: We should check that TiKV is not scaled in here.

File: tests/e2e/tidbcluster/serial.go [hunk 5]
```diff
@@ -713,4 +716,290 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+					return false, nil
+				}
+				if len(tc.Status.TiKV.Stores) != 3 {
+					klog.Info("tikv's store haven't auto-scale to 3")
+					return false, nil
+				}
+				if tc.Annotations != nil && len(tc.Annotations) > 0 {
+					_, ok := tc.Annotations[label.AnnTiKVAutoScalingOutOrdinals]
+					if ok {
+						klog.Infof("tikv auto-scale out annotation is stil existed")
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/serial.go:881]:
```suggestion
						klog.Infof("tikv auto-scale out annotation still exists")
```

File: tests/e2e/tidbcluster/serial.go [hunk 6]
```diff
@@ -713,4 +716,290 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+				}
+				tac, err = cli.PingcapV1alpha1().TidbClusterAutoScalers(ns).Get(tac.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				if tac.Annotations == nil || len(tac.Annotations) < 1 {
+					klog.Infof("tc haven't marked any annotation")
+					return false, nil
+				}
+				v, ok := tac.Annotations[label.AnnTiKVLastAutoScalingTimestamp]
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/serial.go:893]: This timestamp will not be the one of the last scale out?

File: tests/e2e/tidbcluster/serial.go [hunk 7]
```diff
@@ -713,4 +716,290 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+					klog.Infof("tac haven't marked any annotation")
+					return false, nil
+				}
+				secondTs, err := strconv.ParseInt(v, 10, 64)
+				if err != nil {
+					return false, err
+				}
+				if secondTs-firstScaleTimestamp < 100 {
+					klog.Infof("tikv second scale's interval isn't meeting the interval requirement")
+					return false, nil
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/serial.go:904]: Should return an error here?

Comment (Yisaer) [reply tests/e2e/tidbcluster/serial.go:904]: No,the second auto-scaling could haven't happened yet here.

Comment (DanielZhangQD) [reply tests/e2e/tidbcluster/serial.go:904]: If scale in happens here, suggest returning error instead of waiting for the timeout.

Comment (Yisaer) [reply tests/e2e/tidbcluster/serial.go:903]: updated.

Comment (Yisaer) [reply tests/e2e/tidbcluster/serial.go:903]: updated.

File: tests/e2e/tidbcluster/serial.go [hunk 8]
```diff
@@ -899,10 +899,12 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 				if err != nil {
 					return false, err
 				}
-				if secondTs-firstScaleTimestamp < 100 {
-					klog.Infof("tikv second scale's interval isn't meeting the interval requirement")
+				if secondTs == firstScaleTimestamp {
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/serial.go:902]: Suggest adding log here so that we can know if TiKV does not scale before timeout.

Comment (Yisaer) [reply tests/e2e/tidbcluster/serial.go:902]: updated.

File: tests/e2e/tidbcluster/serial.go [hunk 9]
```diff
@@ -713,4 +716,296 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
+				if tc.Annotations == nil || len(tc.Annotations) < 1 {
+					klog.Infof("tc haven't marked any annotation")
+					return false, nil
+				}
+				tac, err = cli.PingcapV1alpha1().TidbClusterAutoScalers(ns).Get(tac.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				if tac.Annotations == nil || len(tac.Annotations) < 1 {
+					klog.Infof("tac haven't marked any annotation")
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/serial.go:811]: Some use `Infof`, some use `Info`, should they be unified?

**Pull request merged**